authentication/subscription/service/oso_subscription_service.go

```diff
@@ -0,0 +1,138 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/fabric8-services/fabric8-auth/application/service"
+	"github.com/fabric8-services/fabric8-auth/application/service/base"
+	servicecontext "github.com/fabric8-services/fabric8-auth/application/service/context"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/rest"
+	"github.com/fabric8-services/fabric8-auth/token"
+	"golang.org/x/oauth2"
+	"net/http"
+
+	autherrors "github.com/fabric8-services/fabric8-auth/errors"
+)
+
+const signUpNeededStatus = "signup_needed"
+
+type OSOSubscriptionServiceConfiguration interface {
+	token.TokenConfiguration
+	GetOSORegistrationAppURL() string
+	GetOSORegistrationAppAdminUsername() string
+	GetOSORegistrationAppAdminToken() string
+}
+
+type subscriptions struct {
+	Subscriptions []Subscription `json:"subscriptions"`
+}
+
+type Subscription struct {
+	Status string `json:"status"`
+	Plan   Plan   `json:"plan"`
+}
+
+type Plan struct {
+	Service Service `json:"service"`
+}
+
+type Service struct {
+	APIURL string `json:"api_url"`
+}
+
+type osoSubscriptionServiceImpl struct {
+	base.BaseService
+	config       OSOSubscriptionServiceConfiguration
+	tokenManager token.Manager
+	httpClient   rest.HttpClient
+}
+
+func NewOSOSubscriptionService(context servicecontext.ServiceContext, config OSOSubscriptionServiceConfiguration) service.OSOSubscriptionService {
+	tokenManager, err := token.NewManager(config)
+	if err != nil {
+		log.Panic(nil, map[string]interface{}{
+			"err": err,
+		}, "failed to create token manager")
+	}
+
+	return &osoSubscriptionServiceImpl{
+		BaseService:  base.NewBaseService(context),
+		config:       config,
+		tokenManager: tokenManager,
+		httpClient:   http.DefaultClient,
+	}
+}
+
+// LoadOSOSubscriptionStatus loads a subscription status from OpenShift Online Registration app
+func (s *osoSubscriptionServiceImpl) LoadOSOSubscriptionStatus(ctx context.Context, token oauth2.Token) (string, error) {
```

Comment:
Might wanna consider add `Ban(..)` method too later on. But, the OSO Ban API calls Auth's deprovision API as well :)